can/interfaces/pcan/pcan.py

```diff
@@ -437,11 +456,26 @@ def set_device_number(self, device_number):
             return False
         return True
 
+    def _read_message(self):
+        """Read a message using Basic API.
+
+        :return: a tuple containing the current CAN status, the received
+            message and the timestamp
+        :rtype: tuple
+        """
+        if self.fd:
+            return self.m_objPCANBasic.ReadFD(self.m_PcanHandle)
+
+        return self.m_objPCANBasic.Read(self.m_PcanHandle)
+
     def _recv_internal(self, timeout):
 
-        if HAS_EVENTS:
+        if HAS_EVENTS and IS_WINDOWS:
             # We will utilize events for the timeout handling
             timeout_ms = int(timeout * 1000) if timeout is not None else INFINITE
+        elif HAS_EVENTS and IS_LINUX:
+            # We will utilize events for the timeout handling
+            timeout_s = timeout if timeout != None else None
```

Comment:
This seems redundant

Reply:
I don't think so select accept timeout in seconds or None.
On windows WaitForSingleObject expects a timeout in ms